ai_agents/personal_finance_agent/app.py

```
@@ -0,0 +1,153 @@
+            with tempfile.NamedTemporaryFile(delete=False, suffix=".csv") as tmp:
+                tmp.write(uploaded.getvalue())
+                temp_path = tmp.name
+            try:
+                summary = ingest_csv(temp_path, replace=True)
+                st.session_state.data_loaded = True
+                st.session_state.import_summary = summary
+                st.success(
+                    f"Imported {summary['rows_imported']} transactions "
+                    f"({summary['date_range']})."
+                )
+            except Exception as exc:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)